packages/vite/src/node/http.ts

```diff
@@ -5,6 +5,7 @@ import type {
   OutgoingHttpHeaders as HttpServerHeaders
 } from 'http'
 import type { ServerOptions as HttpsServerOptions } from 'https'
+import { promises as dns } from 'dns'
```

Comment:
Here I'm using `dns.promises` since `dns/promises` is supported from Node 15+.
https://nodejs.org/dist/latest-v16.x/docs/api/dns.html#dns-promises-api